src/main/java/io/reactivesw/category/infrastructure/validator/CategoryNameValidator.java

```diff
@@ -64,16 +64,16 @@ public static void validateNull(CategoryDraft draft) {
   public static void validateEqual(LocalizedString name, List<Category> sameRootCategories) {
     List<LocalizedStringValue> categoryNames =
         CategoryUtils.getAllCategoryNames(sameRootCategories);
-    Map<String, String> localizedName = name.getLocalized();
-    for (Map.Entry entry : localizedName.entrySet()) {
-      String key = entry.getKey().toString();
-      String value = entry.getValue().toString();
-      for (LocalizedStringValue categoryName : categoryNames) {
-        if (key.equals(categoryName.getLanguage()) && value.equals(categoryName.getText())) {
-          LOG.debug("Can not create category with same name: {}, key: {}.", value, key);
-          throw new AlreadyExistException("Can not create category with same name.");
-        }
+
+    name.getLocalized().entrySet().stream().forEach(localizedName -> {
+      if (categoryNames.stream().anyMatch((categoryName) ->
```

Comment:
can move "(categoryName) -> localizedName.getKey().toString().equals(categoryName.getLanguage()) && localizedName .getValue().toString().equals(categoryName.getText())" to a predicate?